src/NuGet.Core/NuGet.ProjectModel/LockFile/AssetsLogMessage.cs

```diff
@@ -59,5 +62,46 @@ public AssetsLogMessage(LogLevel logLevel, NuGetLogCode errorCode, string errorS
         {
         }
 
+        public bool LogEquals(IAssetsLogMessage other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (Level == other.Level &&
+                Code == other.Code &&
+                WarningLevel == other.WarningLevel &&
+                StartLineNumber == other.StartLineNumber &&
+                EndLineNumber == other.EndLineNumber &&
+                StartColumnNumber == other.StartColumnNumber &&
+                EndColumnNumber == other.EndColumnNumber &&
+                ((Message == null && other.Message == null) || 
+                    Message.Equals(other.Message, StringComparison.Ordinal)) &&
+                ((ProjectPath == null && other.ProjectPath == null) || 
+                    ProjectPath.Equals(other.ProjectPath, StringComparison.Ordinal)) &&
+                ((FilePath == null && other.FilePath == null) || 
+                    FilePath.Equals(other.FilePath, StringComparison.Ordinal)) &&
+                ((LibraryId == null && other.LibraryId == null) || 
+                    LibraryId.Equals(other.LibraryId, StringComparison.Ordinal)))             
+            {
+
+                if (TargetGraphs != null && other.TargetGraphs != null)
```

Comment:
Use ``EqualityUtility.SequenceEqualWithNullCheck``  these extra helpers were added to cover these scenarios for LockFile